lib/crypto/crypto.js

```diff
@@ -257,6 +257,32 @@ class Crypto {
       .join('.')
   }
 
+  /**
+   * Encrypt a private key using a keypair.
+   *
+   * @param {string} privateKey
+   * @param {KeyPair} encryptionKeyPair
+   *
+   * @return {Promise<string>}
+   */
+  async encryptPrivateKey(privateKey, encryptionKeyPair) {
+    let rawPrivateKey = this.platform.b64URLDecode(privateKey)
+    let publicEncKey = this.platform.b64URLDecode(encryptionKeyPair.publicKey)
```

Comment:
Do we do this in the encrypt eak? A lot of times you are going to have a private key and public key from different key pairs, so its probably a good idea to split them into to separate parameters. (example: If I write a membership key for your client, I'll use my private key, and your public key).